src/Shared/SignalVersionHelper.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Reflection;
```

Comment:
naïve question: Does this have any implication for AOT?  Does AOT support getting assembly info?

Reply:
I do not see any compilation warnings. Some instrumentation packages are included into AOT app.
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/e39306fc82c2e8f0ef9cde54b3d2701828add459/test/OpenTelemetry.AotCompatibility.TestApp/OpenTelemetry.AotCompatibility.TestApp.csproj#L18-L22